src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key];
+
+    const keyString = String(keyValue);
+
+    if (!result[keyString]) {
+      result[keyString] = [];
+    }
+    result[keyString].push(item);
+
+    return result;
+  }, {} as GroupsMap<T>);
```

Comment:
The initial value for 'reduce' is being cast to 'GroupsMap<T>', which works, but it might be clearer to provide an actual initial value that conforms to 'GroupsMap<T>' without casting.